demos/immersed_fem/immersed_fem.py.rst

```diff
@@ -0,0 +1,115 @@
+element size at these corners (which we set to 0.5). ::
+
+  Point(1) = {-6,  2, 0, 0.5};
+  Point(2) = {-6, -2, 0, 0.5};
+  Point(3) = { 6, -2, 0, 0.5};
+  Point(4) = { 6,  2, 0, 0.5};
+
+Then, we define 5 points to describe a circle. ::
+
+  Point(5) = {   0,    0, 0, 0.1};
+  Point(6) = { 0.5,    0, 0, 0.1};
+  Point(7) = {-0.5,    0, 0, 0.1};
+  Point(8) = {   0,  0.5, 0, 0.1};
+  Point(9) = {   0, -0.5, 0, 0.1};
+
+Then, we create 8 edges: 4 for the rectangle and 4 for the circle.
+Recall that the Gmsh command ``Circle`` requires the arc to be
+strictly smaller than :math:`\pi`. ::
+
+  Line(1) = {1, 4};
+  Line(2) = {4, 3};
+  Line(3) = {3, 2};
+  Line(4) = {2, 1};
+  Circle(5) = {8, 5, 6};
+  Circle(6) = {6, 5, 9};
+  Circle(7) = {9, 5, 7};
+  Circle(8) = {7, 5, 8};
+
+Then, we glue together the rectangle edges and, separately, the circle edges.
+Note that ``Line``, ``Circle``, and ``Curve Loop`` (as well as ``Physical Curve`` below)
+are all curves in Gmsh and must possess a unique tag. ::
+
+  Curve Loop( 9) = {1, 2, 3, 4};
+  Curve Loop(10) = {8, 5, 6, 7};
+
+Then, we define two plane surfaces: the rectangle without the disc first, and the disc itself then. ::
+
+  Plane Surface(1) = {9, 10};
+  Plane Surface(2) = {10};
+
+Finally, we group together some edges and define ``Physical`` entities, which
+can be accessed in Firedrake. ::
+
+  Physical Curve("HorEdges", 11) = {1, 3};
+  Physical Curve("VerEdges", 12) = {2, 4};
+  Physical Curve("Circle", 13) = {8, 7, 6, 5};
+  Physical Surface("PunchedDom", 3) = {1};
+  Physical Surface("Disc", 4) = {2};
+
+For simplicity, we have gathered all this commands in the file
+`immersed_domain.geo <immersed_domain.geo>`__. To generate a mesh using this file,
+you can type the following command in the terminal::
+
+    gmsh -2 punched_domain.geo -format msh2
+
+To illustrate how to access all these features within Firedrake,
+we consider the following interface problem. Denoting by
+:math:`\Omega` the filled rectangle and by :math:`D` the disc,
+we seek a function :math:`u\in H^1_0(\Omega)` such that
+
+.. math::
+
+   -\nabla \cdot (\sigma \nabla  u) + u = 5 \quad \textrm{in } \Omega
+
+where :math:`\sigma = 1` in :math:`\Omega \setminus D` and :math:`\sigma = 2`
+in :math:`D`. Since :math:`\sigma` attains different values across :math:`\partial D`,
+we need to prescribe the behavior of :math:`u` across this interface. This is
+implicitely done by imposing :math:`u\in H^1_0(\Omega)`: the function :math:`u` must be continuous
+across :math:`\partial \Omega`. This allows us to employ Lagrangian finite elements
+to approximate :math:`u`. However, we also need to specify the the jump
+of :math:`\sigma \nabla u \cdot \vec{n}` on :math:`\partial D`. This term arises
+naturally in the weak formulation of the problem under consideration. In this demo
+we simply set
+
+.. math::
+
+   [\sigma \nabla u \cdot \vec{n}] = 3 \quad \textrm{on}\ \partial D
```

Comment:
Use `\\llbracket` and `\\rrbracket` for jump

Reply:
updated, thanks